crates/net/discv4/src/config.rs

```diff
@@ -54,6 +54,8 @@ pub struct Discv4Config {
     pub enable_eip868: bool,
     /// Additional pairs to include in The [`Enr`](enr::Enr) if EIP-868 extension is enabled <https://eips.ethereum.org/EIPS/eip-868>
     pub additional_eip868_rlp_pairs: HashMap<Vec<u8>, Bytes>,
+    /// disables discovery service if set to true.
+    pub disable_discovery: bool,
```

Comment:
let's move to Networkconfig, perhaps even solve this with `Option<DiscocveryConfig>` there